src/ts/components/subtitlesettings/fontopacityselectbox.ts

```diff
@@ -1,37 +1,41 @@
-import {SubtitleSettingSelectBoxConfig, SubtitleSettingSelectBox} from './subtitlesettingselectbox';
+import {SubtitleSettingSelectBox} from './subtitlesettingselectbox';
 import {UIInstanceManager} from '../../uimanager';
-import {StorageUtils} from '../../storageutils';
-import {ColorUtils} from '../../colorutils';
 
 /**
  * A select box providing a selection of different font colors.
  */
 export class FontOpacitySelectBox extends SubtitleSettingSelectBox {
 
-  constructor(config: SubtitleSettingSelectBoxConfig) {
-    super(config);
-  }
-
   configure(player: bitmovin.PlayerAPI, uimanager: UIInstanceManager): void {
     super.configure(player, uimanager);
 
-    this.addItem('0.25', '25%');
-    this.addItem('0.50', '50%');
-    this.addItem('0.75', '75%');
-    this.addItem('1', '100%');
+    this.addItem(null, 'default');
+    this.addItem('100', '100%');
+    this.addItem('75', '75%');
+    this.addItem('50', '50%');
+    this.addItem('25', '25%');
 
-    this.selectItem('1');
+    this.onItemSelected.subscribe((sender, key: string) => {
+      this.settingsManager.fontOpacity.value = key;
 
-    if (StorageUtils.hasLocalStorage()) {
-      let color = window.localStorage.getItem('fontColor');
-      if (color != null) {
-        let col = ColorUtils.colorFromCss(color, ColorUtils.foreground);
-        this.selectItem(col.a.toString());
+      // Color and opacity go together, so we need to...
```

Comment:
Opacity and color always need to be set together. Due to CSS `rgba` coloring, it is not possible to *only* set the opacity (opacity cannot be set separately in CSS and we cannot "inherit" the base color) *or* the color (color cannot be set separately from the opacity).
Therefore, I select a default opacity when a color is selected, or the other way round. Afterwards, then can be set independently, i.e. color is not reset to default every time an opacity is set. Or did you make an opposite observation?